docs/contributing.md

```diff
@@ -38,7 +38,13 @@ use `jj split` to do it if you didn't realize ahead of time how it should be
 split up. Include tests and documentation in the same commit as the code they
 test and document. The commit message should describe the changes in the commit;
 the PR description can even be empty, but feel free to include a personal
-message.
+message. We write commit messages in a affected component style and don't use 
+[conventional commits](www.conventionalcommits.org/en/v1.0.0/), this means if 
+you modified a command in the cli, use it's name as the topic, e.g 
+`next/prev: <your-modification>` or `conflicts: <your-modification>`. We don't
+currently have a specific guidelines on what to write in the topic field, but 
+the reviewers will help you provide a topic if you have difficulties choosing 
+it.
```

Comment:
@ilyagr PTAL

Reply:
nit:
a affected → an affected
, this means → . This means
cli → CLI
it's name → its name

Thanks for initiating this PR! I've always wondered about the 'convention' around here.

Reply:
It seems like it would be useful if someone could state the benefit of having a convention. Maybe it's what you're kind of saying, @cenviity, that having written down that there's convention means that people don't have to wonder whether there is a convention. If that's the reason, I suppose an alternative is to explicitly state that you don't need to follow any convention (if that's the consensus).

I personally usually use a `<topic>: ` prefix. I think I do that because I'm used to it from the Git and Mercurial projects, but I'm not sure anyone has explained to me why I was supposed to do it 🤷

Reply:
Yes, mentioning this 'convention' in the contribution guidelines would stop people from having to wonder indeed – even if it's the fact that, ironically, a convention of whether to include a topic does _not_ exist. That's where the lowering of cognitive load comes in.

I don't feel too strongly one way or the other re: including a topic, though I can see one benefit of including a topic is that it helps with quickly glancing through the commit log after fetching repo changes.

Reply:
Sorry it took me a while to get back to this.

Apart from cenviity 's helpful nits (https://github.com/martinvonz/jj/pull/4356#discussion_r1753108052) above, this text looks fine to me. Thank you for the edits! I'll approve this based on the fact that I think it describes our status-quo well.

> having written down that there's convention means that people don't have to wonder whether there is a convention. If that's the reason, I suppose an alternative is to explicitly state that you don't need to follow any convention (if that's the consensus).

This is a good reason to write down something like this.

> I personally usually use a <topic>: prefix. I think I do that because I'm used to it from the Git and Mercurial projects, but I'm not sure anyone has explained to me why I was supposed to do it 🤷

I don't necessarily have an answer, but I generally like the topic prefix. It allows me to quickly look at the list of recent `jj` commits and judge what kind of things have changed and (for example) how likely I am to see conflicts if I rebase.

Reply:
Good point about using it for scanning the changelog. I have sometimes mixed topics in a PR (like https://github.com/martinvonz/jj/pull/4300/commits). Maybe I should try to more consistently use the same topic in a PR/series. I'll try to remember to do that next time.

Reply:
I also think that stating that we have a convention is beneficial for contributors, instead of stating that it is a free for all. And the `topic:` convention will scale nicely to further projects like the Web UI for `jj web`, native server and the different CR providers.